Gordon360/Controllers/RecIM/SportsController.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Models.ViewModels.RecIM;
+using Gordon360.Services.RecIM;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Gordon360.Controllers.RecIM
+{
+    [Route("api/recim/[controller]")]
+    [AllowAnonymous]
+    public class SportsController : GordonControllerBase
+    {
+        private readonly ISportService _sportService;
+
+        public SportsController(ISportService sportService)
+        {
+            _sportService = sportService;
+        }
+
+        [HttpGet]
+        [Route("")]
+        public ActionResult<IEnumerable<SportViewModel>> GetSports()
+        {
+            var res = _sportService.GetSports();
+            return Ok(res);
+        }
+
+        [HttpGet]
+        [Route("{sportID}")]
+        public ActionResult<SportViewModel> GetSportByID(int sportID)
+        {
+            var res = _sportService.GetSportByID(sportID);
+            return Ok(res);
+        }
+      
+        [HttpPatch]
+        [Route("{sportID}")]
+        public async Task<ActionResult> UpdateSport(int sportID, SportViewModel updatedSport)
+        {
+            await _sportService.UpdateSport(updatedSport);
```

Comment:
You should pass the sportID to the service method and use that to find the sport that you will update in the service (remember not to match up the database row using `updatedSport.sportID`)
```suggestion
            await _sportService.UpdateSport(sportID, updatedSport);
```

Reply:
will update all these things in the next PR